build/teamcity-release-notes.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+cockroach_ref=$(git describe --tags --exact-match 2>/dev/null || git rev-parse HEAD)
```

Comment:
TeamCity always passes the `BUILD_VCS_NUMBER` variable in the environment to scripts it calls. It contains the SHA of the commit that triggered the build. I suggest using that instead of getting the latest tag from git. (This will also make testing locally easier since you'll be able to set this to any commit you want to test against.)